### ISSUE TITLE
Colored error messages

### ISSUE THREAD

**Issue by DemiMarie** [rust-lang/rust-mode#184]
Emacs should display error messages using the same colors emitted by the compiler.

**Comment by Rufflewind** [comment]
Agreed.  In particular, I also propose displaying warnings and errors in their respective colors (including the filename and line numbers), since I tend to rely on the color to tell whether the diagnostic is important or not.
